src/community-toolbox.js

```diff
@@ -39,6 +39,130 @@ CommunityToolbox = function CommunityToolbox(org, repo) {
        .then(callback);
   }
 
+  function fetchRepoContributors(org, repo) {
+
+    var arr = [];
+    api.Repositories
+           .getRepoContributors(org, repo, {method: "HEAD", qs: { sort: 'pushed', direction: 'desc', per_page: 100 } })
+           .then(function(contribData) {
+             var headers = contribData;
+             if (headers.hasOwnProperty("link")) {
+                 var parsed = parse(headers['link']);
+                 totalPages = parseInt(parsed.last.page);
+             } else {
+                 totalPages = 1;
+             }
+             return totalPages;
+           })
+           .then((totalPages) => {
+              for(var i = 1; i <= totalPages; i++) {
+                api.Repositories
+                .getRepoContributors(org, repo, { method:"GET", qs: { sort: 'pushed', direction: 'desc', per_page: 100, page:i } })
+                .then(function(contributors) {
+                    if (contributors != null || contributors.length > 0) {
+                      contributors.map((contributor, i) => arr.push(contributor));
+                    }
+                });
+              }
+           })
+
+
+  return new Promise((resolve, reject) => {
```

Comment:
so you did end up using promises? cool!

Reply:
Yeah, I used them but things didn't happen the way I thought, so I ended up putting up `setTimeout` everywhere...(async/await are not allowed by grunt-browserify, it is giving error while `grunt build`)